core/trino-main/src/main/java/io/trino/sql/planner/planprinter/anonymize/Anonymizer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.planprinter.anonymize;
+
+import io.trino.execution.TableInfo;
+import io.trino.metadata.IndexHandle;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.metadata.TableExecuteHandle;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.PartitioningHandle;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.tree.Expression;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.sql.planner.Partitioning.ArgumentBinding;
+import static io.trino.sql.planner.plan.StatisticsWriterNode.WriteStatisticsTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.WriterTarget;
+
+public interface Anonymizer
```

Comment:
Sending an anonymised plan into PlanPrinter probably would be a cleaner approach.
Could you point out the blockers to taking that approach ?

Reply:
> Sending an anonymised plan into PlanPrinter probably would be a cleaner approach.

There are a few issues with this approach. If these are acceptable then we can implement this way.

- For things like `ConnectorTableHandle`, `ConnectorInsertTableHandle`, `ColumnHandle` etc, we need to create some anonymised version of these classes where toString print the anonymized information. IMO this could get complicated.

- If we add a generic way to anonymize `PlanNode` then whenever we add a new field to any of the `PlanNode` implementations, we have to anonymize that part even if it isn't used in `PlanPrinter`. This seems unnecessary.

WDYT? @raunaqmorarka @martint